src/fmu/dataio/preprocessed.py

```diff
@@ -0,0 +1,226 @@
+        is not present, None will be returned.
+        """
+        if objmetafile.exists():
+            with open(objmetafile) as stream:
+                return yaml.safe_load(stream)
+        return None
+
+    def _get_relative_export_path(self, existing_path: Path) -> Path:
+        """Get an updated relative_path from an existing path."""
+
+        existing_share_folder = ShareFolder.PREPROCESSED.value + "/"
+        existing_subfolders_and_filename = str(existing_path).rsplit(
+            existing_share_folder, maxsplit=1
+        )[-1]
+
+        if not self.is_observation:
+            return Path(ShareFolder.RESULTS.value) / existing_subfolders_and_filename
+        return Path(ShareFolder.OBSERVATIONS.value) / existing_subfolders_and_filename
+
+    @staticmethod
+    def _check_md5sum_consistency(
+        checksum_md5_file: str, checksum_md5_meta: str
+    ) -> None:
+        """Check if the md5sum for the file is equal to the one in the metadata"""
+        if checksum_md5_file != checksum_md5_meta:
+            warnings.warn(
+                "The preprocessed file seem to have been modified since it was "
+                "initially exported. You are adviced to re-create the preprocessed "
+                "data to prevent mismatch between the file and its metadata."
+            )
+
+    def _get_meta_file(self, objfile: Path, checksum_md5: str) -> meta.File:
+        """Return a meta.File model with updated paths and checksum_md5"""
+        relative_path = self._get_relative_export_path(existing_path=objfile)
+        return meta.File(
+            absolute_path=self.casepath / relative_path,
+            relative_path=relative_path,
+            checksum_md5=checksum_md5,
+        )
+
+    def _get_updated_metadata(self, meta_existing: dict, objfile: Path) -> dict:
+        """
+        Update the existing metadata with updated fmu/file/tracklog info:
+        - The 'fmu' block will be added
+        - The 'file' block will be updated with new paths.
+        - The 'tracklog' block will be extended with a new event tagged "merged".
+
+        A simple consistency check will be run to detect if the file has been
+        modified since it was initially exported.
+        """
+        assert self._fmudata is not None
+
+        checksum_md5_file = md5sum(objfile)
+        if checksum_md5_meta := meta_existing["file"].get("checksum_md5"):
+            self._check_md5sum_consistency(checksum_md5_file, checksum_md5_meta)
+
+        # remove legacy '_preprocessed' key if present
+        if "_preprocessed" in meta_existing:
+            del meta_existing["_preprocessed"]
+
+        meta_existing["fmu"] = self._fmudata.get_metadata()
+        meta_existing["file"] = self._get_meta_file(objfile, checksum_md5_file)
+
+        # update the tracklog block
+        tracklog_entry = generate_meta_tracklog(event="merged")
+        meta_existing["tracklog"].extend(tracklog_entry)
```

Comment:
I think we should consider a class or set of functions to do these operations, but not in this PR #662